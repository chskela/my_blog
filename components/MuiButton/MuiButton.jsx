import React from "react";
import cn from "classnames";
import Link from "next/link";
import PropTypes from "prop-types";

import classes from "./class.module.css";

const MuiButton = React.memo(
  ({ label, type = "button", className = "", onClick = () => {} }) => (
    <button
      type={type}
      onClick={(event) => onClick(event)}
      className={cn(classes.button, className)}
    >
      {label}
    </button>
  )
);

MuiButton.propTypes = {
  href: PropTypes.string,
  type: PropTypes.oneOf(["reset", "button", "submite"]),
  className: PropTypes.string,
  onClick: PropTypes.func,
};

export default MuiButton;
