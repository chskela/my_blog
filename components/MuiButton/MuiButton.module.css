.button {
  font-weight: 400;
  line-height: var(--size-6);
  border-radius: 10px;
  box-sizing: border-box;
  font-size: var(--size-5);
  color: var(--color-secondary);
  padding: var(--size-3) var(--size-8);
  background-color: var(--color-primary);
  font-family: "Libre Baskerville", serif;
  border: 2px solid var(--color-secondary);
}
