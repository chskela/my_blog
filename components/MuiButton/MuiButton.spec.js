import React from "react";
import renderer from "react-test-renderer";
import MuiButton from "./MuiButton";

test("MuiButton cteated", () => {
  const component = renderer.create(<MuiButton label={"Home"} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
