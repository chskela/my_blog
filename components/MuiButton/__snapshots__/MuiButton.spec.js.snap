// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MuiButton cteated 1`] = `
<button
  className="button"
  onClick={[Function]}
  type="button"
>
  Home
</button>
`;
